atomisticparsers/h5md/parser.py

```diff
@@ -0,0 +1,787 @@
+
+            for path in observable_paths:
+                observable = self._data_parser.hdf5_getter(observables_group, path)
+                observable_type = self._data_parser.hdf5_getter(observables_group, path).attrs.get('type')
+                observable_name = path.split('.')[0]
+                observable_label = '-'.join(path.split('.')[1:]) if len(path.split('.')) > 1 else ''
+                if observable_name not in self._observable_info[observable_type].keys():
+                    self._observable_info[observable_type][observable_name] = {}
+                self._observable_info[observable_type][observable_name][observable_label] = {}
+                for key in observable.keys():
+                    observable_attribute = self._data_parser.hdf5_getter(observable, key)
+                    if type(observable_attribute) == h5py.Group:
+                        self.logger.warning('Group structures within individual observables not supported. ' + key + ' values will not be stored.')
+                        continue
+                    self._observable_info[observable_type][observable_name][observable_label][key] = observable_attribute
+        return self._observable_info
+
+    def get_atomsgroup_fromh5md(self, nomad_sec, h5md_sec_particlesgroup):
+        for i_key, key in enumerate(h5md_sec_particlesgroup.keys()):
+            particles_group = {group_key: self._data_parser.hdf5_getter(h5md_sec_particlesgroup[key], group_key) for group_key in h5md_sec_particlesgroup[key].keys()}
+            sec_atomsgroup = nomad_sec.m_create(AtomsGroup)
+            sec_atomsgroup.type = particles_group.pop('type', None)
+            sec_atomsgroup.index = i_key
+            sec_atomsgroup.atom_indices = particles_group.pop('indices', None)
+            sec_atomsgroup.n_atoms = len(sec_atomsgroup.atom_indices) if sec_atomsgroup.atom_indices is not None else None
+            sec_atomsgroup.is_molecule = particles_group.pop('is_molecule', None)
+            sec_atomsgroup.label = particles_group.pop('label', None)
+            sec_atomsgroup.composition_formula = particles_group.pop('formula', None)
+            particles_subgroup = particles_group.pop('particles_group', None)
+            # set the remaining attributes
+            for particles_group_key in particles_group.keys():
+                unit = None
+                val = particles_group.get(particles_group_key)
+                has_units = hasattr(val, 'units') if val is not None else None
+                if has_units:
+                    unit = val.units
+                    val = val.magnitude
+                sec_atomsgroup.x_h5md_parameters.append(ParamEntry(kind=particles_group_key, value=val, unit=unit))
+            # get the next atomsgroup
+            if particles_subgroup:
+                self.get_atomsgroup_fromh5md(sec_atomsgroup, particles_subgroup)
+
+    def check_metainfo_for_key_and_Enum(self, metainfo_class, key, val):
```

Comment:
I think this is not necessary, base class will include x_ quantities if extend_base_section=True.

Reply:
ok, yeah, I was definitely not approaching this correctly. I think this part was outdated with how I am now approaching the non-normalized metadata. 

I now added the appropriate definitions in the h5md metainfo file. I still use a function to check the key/val pair (I need to cover the case that the key is valid but value does not match the MEnum list), but you can just see my changes in the next round of reviews.